XCTest/Tests/Routes/LPSetDeviceOrientationRouteTest.m

```diff
@@ -0,0 +1,178 @@
+}
+
+- (void)testRespondsToPOST {
+  BOOL actual = [self.route supportsMethod:@"POST" atPath:nil];
+  expect(actual).to.equal(YES);
+}
+
+- (void)testDoesNotRespondToGET {
+  BOOL actual = [self.route supportsMethod:@"GET" atPath:nil];
+  expect(actual).to.equal(NO);
+}
+
+- (void)testOrientationForStringUpsideDown {
+  NSInteger expected = (NSInteger)UIDeviceOrientationPortraitUpsideDown;
+
+  expect([self.route orientationForString:@"up"]).to.equal(expected);
+  expect([self.route orientationForString:@"top"]).to.equal(expected);
+  expect([self.route orientationForString:@"upside down"]).to.equal(expected);
+}
+
+- (void)testOrientationForStringPortrait {
+  NSInteger expected = (NSInteger)UIDeviceOrientationPortrait;
+
+  expect([self.route orientationForString:@"bottom"]).to.equal(expected);
+  expect([self.route orientationForString:@"down"]).to.equal(expected);
+  expect([self.route orientationForString:@"portrait"]).to.equal(expected);
+}
+
+- (void)testOrientationForStringLandscapeRight {
+  NSInteger expected = (NSInteger)UIDeviceOrientationLandscapeRight;
+
+  expect([self.route orientationForString:@"left"]).to.equal(expected);
+  expect([self.route orientationForString:@"landscape right"]).to.equal(expected);
+}
+
+- (void)testOrientationForStringLandscapeLeft {
+  NSInteger expected = (NSInteger)UIDeviceOrientationLandscapeLeft;
+
+  expect([self.route orientationForString:@"right"]).to.equal(expected);
+  expect([self.route orientationForString:@"landscape left"]).to.equal(expected);
+}
+
+- (void)testOrientationForStringUnexpectedInput {
+  NSInteger expected = (NSInteger)UIDeviceOrientationPortrait;
+
+  expect([self.route orientationForString:@"face down"]).to.equal(expected);
+  expect([self.route orientationForString:@"face up"]).to.equal(expected);
+  expect([self.route orientationForString:@"unknown"]).to.equal(expected);
+  expect([self.route orientationForString:nil]).to.equal(expected);
+}
+
+- (void)testIsValidDeviceOrientationYES {
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationPortrait]).to.equal(YES);
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationPortraitUpsideDown]).to.equal(YES);
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationLandscapeLeft]).to.equal(YES);
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationLandscapeRight]).to.equal(YES);
+}
+
+- (void)testIsValidDeviceOrientationNO {
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationPortrait - 1]).to.equal(NO);
+  expect([self.route isValidUIDeviceOrientation:UIDeviceOrientationLandscapeRight + 1]).to.equal(NO);
+}
+
+- (void)testOrientationWithDictionaryNoValueForKey {
+  NSDictionary *args = @{};
+
+  expect([self.route orientationWithDictionary:args]).to.equal(1);
+}
+
+- (void)testOrientationWithDictionaryStringValue {
+  NSDictionary *args = @{
+                         @"orientation" : @"string"
+                         };
+  NSInteger expected = 10;
```

Comment:
In both cases, I am mocking the output of a method to return _exactly_ what I want.

The goal to test all code paths.  

In particular: 

* "What happens if the orientation is an NSNumber?" and 
* "What happens if the orientation is an NSString?".

I did not want to use an valid orientation because I wanted to make it clear that values are mocked.